tests/test_masking.py

```diff
@@ -0,0 +1,237 @@
+        PACKAGE_NAME,
+        "CLI Tools Testing Project",
+        fake_root_dir / REPO_NAME,
+        kedro_version,
+        fake_root_dir / REPO_NAME / "src",
+    )
+    return metadata
+
+
+class TestCLIMasking:
+    def test_get_cli_structure(self, mocker, fake_metadata):
+        # This test is a replica of `kedro.tests.tools.test_cli.py` tests
+        # This duplication is to maintain backwards compatibility
+        Module = namedtuple("Module", ["cli"])
+        mocker.patch(
+            "kedro.framework.cli.cli.importlib.import_module",
+            return_value=Module(cli=cli),
+        )
+        mocker.patch(
+            "kedro.framework.cli.cli._is_project", return_value=True,
+        )
+        mocker.patch(
+            "kedro.framework.cli.cli.bootstrap_project", return_value=fake_metadata,
+        )
+        kedro_cli = KedroCLI(fake_metadata.project_path)
+        raw_cli_structure = get_cli_structure(kedro_cli, get_help=False)
+        help_cli_structure = get_cli_structure(kedro_cli, get_help=True)
+
+        # raw CLI structure tests
+        assert isinstance(raw_cli_structure, dict)
+        assert isinstance(raw_cli_structure["kedro"], dict)
+
+        for k, v in raw_cli_structure["kedro"].items():
+            assert isinstance(k, str)
+            assert isinstance(v, dict)
+
+        assert sorted(list(raw_cli_structure["kedro"])) == sorted(
+            DEFAULT_KEDRO_COMMANDS
+        )
+
+        # get_help CLI structure tests
+        assert isinstance(help_cli_structure, dict)
+        assert isinstance(help_cli_structure["kedro"], dict)
+
+        for k, v in help_cli_structure["kedro"].items():
+            assert isinstance(k, str)
+            if isinstance(v, dict):
+                for sub_key in v:
+                    assert isinstance(help_cli_structure["kedro"][k][sub_key], str)
+                    assert help_cli_structure["kedro"][k][sub_key].startswith(
+                        "Usage:  [OPTIONS]"
+                    )
+            elif isinstance(v, str):
+                assert v.startswith("Usage:  [OPTIONS]")
+
+        assert sorted(list(help_cli_structure["kedro"])) == sorted(
+            DEFAULT_KEDRO_COMMANDS
+        )
+
+    @pytest.mark.parametrize(
+        "input_dict, expected_output_count",
+        [
+            ({}, 0),
+            ({"a": "foo"}, 2),
+            ({"a": {"b": "bar"}, "c": {"baz"}}, 5),
+            (
+                {
+                    "a": {"b": "bar"},
+                    "c": None,
+                    "d": {"e": "fizz"},
+                    "f": {"g": {"h": "buzz"}},
+                },
+                12,
+            ),
+        ],
+    )
+    def test_recursive_items(self, input_dict, expected_output_count):
+        assert expected_output_count == sum(
+            1 for _ in _recursive_items(dictionary=input_dict)
+        )
+
+    def test_recursive_items_empty(self):
+        assert len(list(_recursive_items({}))) == 0
```

Comment:
Very minor suggestion, feel free to ignore...
```suggestion
        assert not list(_recursive_items({}))
```
or
```suggestion
        assert list(_recursive_items({})) == []
```

Reply:
Not a fan of these - I know what you mean but I would prefer to be explicit here, rather than checking for `false`-y values.